src/ledger/parse/account-delete.ts

```diff
@@ -11,6 +11,10 @@ export type FormattedAccountDelete = {
   // for the recipient of the deleted account's leftover XRP. NB: Ensure that the hosted recipient is
   // able to account for AccountDelete transactions; if not, your balance may not be properly credited.
   destinationTag?: number
+
+  // X-address of an account to receive any leftover XRP after deleting the sending account.
+  // Must be a funded account in the ledger, and must not be the sending account.
+  destinationXAddress: string
```

Comment:
Does this mean you have to supply both the normal address and X Address? Ideally it's either/or:

`normal_destination_address (+ optional_tag) || x_address`

right?

Reply:
right, it should be either/or

Reply:
This particular file is `src/ledger/parse/account-delete.ts` which is used to parse an existing `AccountDelete` transaction from the ledger. When you retrieve an `AccountDelete` tx and it's parsed by ripple-lib, you should always get both the classic address and the X-address. So I think this code is correct

Reply:
I'm not going to block this PR from going in, but that feels like a *really* weird API that isn't intuitive to use. I'd recommend just choosing one or the other (probably X-Addresses) or creating two sets of APIs. 

My rationale is that giving users too much information, or too many options, leads to confusion and greater chances of mistakes and / or bugs.

Reply:
👍 Let's try to move to X-address-only in ripple-lib v2.0